src/test/script_tests.cpp

```diff
@@ -1925,8 +1928,10 @@ BOOST_AUTO_TEST_CASE(cat_dup_test)
             witVerifyScript.push_back(OP_CAT);
             int expectedErr = SCRIPT_ERR_OK;
             unsigned int catedStackElementSize = witData.at(0).size()*pow(2, dups);
-            if (catedStackElementSize > MAX_SCRIPT_ELEMENT_SIZE || elementSize > MAX_SCRIPT_ELEMENT_SIZE)
+            if (catedStackElementSize > MAX_SCRIPT_ELEMENT_SIZE || elementSize > MAX_SCRIPT_ELEMENT_SIZE){
                 expectedErr = SCRIPT_ERR_PUSH_SIZE;
+                break;
```

Comment:
can you remind me why we break here?

Reply:
The intention is break after we hit the first SCRIPT_ERR_PUSH_SIZE; error so that we don't waste time testing after we already past the edge case.  This was because my test was taking 40 seconds when rest the entire test suite was only taking 6 seconds.

However my code has a bug. It should be, the break hits too early

```cpp
for (unsigned int dups = 1; dups <= maxDupsToCheck; dups++) {
            witVerifyScript.push_back(OP_DUP);
            witVerifyScript.push_back(OP_CAT);
            int expectedErr = SCRIPT_ERR_OK;
            unsigned int catedStackElementSize = witData.at(0).size()*pow(2, dups);
            if (catedStackElementSize > MAX_SCRIPT_ELEMENT_SIZE || elementSize > MAX_SCRIPT_ELEMENT_SIZE){
                expectedErr = SCRIPT_ERR_PUSH_SIZE;
            }
            DoTapscriptTest(witVerifyScript, witData, "CAT DUP test", expectedErr);
            If ( expectedErr == SCRIPT_ERR_PUSH_SIZE)
                 break;
        }
```